src/rules/require-to-throw-message.ts

```diff
@@ -0,0 +1,50 @@
+import {
+  ModifierName,
+  createRule,
+  isExpectCall,
+  parseExpectCall,
+} from './utils';
+
+export default createRule({
+  name: __filename,
+  meta: {
+    docs: {
+      category: 'Best Practices',
+      description: 'Require a message for `toThrow()`',
+      recommended: false,
+    },
+    messages: {
+      requireRethrow: 'Add an error message to {{ propertyName }}()',
+    },
+    type: 'suggestion',
+    schema: [],
+  },
+  defaultOptions: [],
+  create(context) {
+    return {
+      CallExpression(node) {
+        if (!isExpectCall(node)) {
+          return;
+        }
+
+        const { matcher, modifier } = parseExpectCall(node);
+
+        if (
+          matcher &&
+          matcher.arguments &&
+          matcher.arguments.length === 0 &&
+          ['toThrow', 'toThrowError'].includes(matcher.name) &&
+          (!modifier ||
+            !(modifier.name === ModifierName.not || modifier.negation))
+        ) {
+          // Look for `toThrow` calls with no arguments.
+          context.report({
+            messageId: 'requireRethrow', // todo: rename to 'addErrorMessage'
```

Comment:
I wouldn't object if you wanted to do these `todo`s for me 😉

Reply:
Would that be a breaking change? I am not too familiar how `messageId` is used.

Reply:
No, it's not breaking - `messageId` is just used internally by `eslint` to make it easier to test, [so you don't have to write things out multiple times](https://eslint.org/docs/developer-guide/working-with-rules#messageids).

Reply:
I've updated this PR to include the changes in TODO. Please verify that the `context.report.data.propertyName` -> `context.report.data.matcherName` was wanted.